plugins/parsers/avro/parser_test.go

```diff
@@ -0,0 +1,269 @@
+package avro
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/linkedin/goavro/v2"
+	"github.com/stretchr/testify/require"
+)
+
+var DefaultTime = func() time.Time {
+	return time.Unix(3600, 0)
+}
+
+func TestBasicAvroMessage(t *testing.T) {
```

Comment:
Can you maybe convert your test-cases to the scheme used in [json_v2](https://github.com/influxdata/telegraf/tree/master/plugins/parsers/json_v2/testdata) or (partially) in [xpath](https://github.com/influxdata/telegraf/tree/master/plugins/parsers/xpath/testcases)? This would ease adding more test-cases and give people a direct configuration example to play with.

Reply:
I'll do that as a separate commit, in that that's going to take understanding more of the tests than I have so far (that's all inherited from the previous PR).

Reply:
Ok, but please take a look as this allows for fast addition of new test-cases by more or less only copying in the user's config and input data.

Reply:
I have done an initial implementation of this.  The rest of the stuff you asked for is much easier.